homeassistant/components/zwave_js/trigger.py

```diff
@@ -2,45 +2,17 @@
 
 from __future__ import annotations
 
-from homeassistant.const import CONF_PLATFORM
-from homeassistant.core import CALLBACK_TYPE, HomeAssistant
-from homeassistant.helpers.trigger import (
-    TriggerActionType,
-    TriggerInfo,
-    TriggerProtocol,
-)
-from homeassistant.helpers.typing import ConfigType
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.trigger import Trigger
 
 from .triggers import event, value_updated
 
 TRIGGERS = {
-    "value_updated": value_updated,
-    "event": event,
+    event.PLATFORM_TYPE: event.EventTrigger,
+    value_updated.PLATFORM_TYPE: value_updated.ValueUpdatedTrigger,
 }
 
 
-def _get_trigger_platform(config: ConfigType) -> TriggerProtocol:
-    """Return trigger platform."""
-    platform_split = config[CONF_PLATFORM].split(".", maxsplit=1)
-    if len(platform_split) < 2 or platform_split[1] not in TRIGGERS:
-        raise ValueError(f"Unknown Z-Wave JS trigger platform {config[CONF_PLATFORM]}")
-    return TRIGGERS[platform_split[1]]
-
-
-async def async_validate_trigger_config(
-    hass: HomeAssistant, config: ConfigType
-) -> ConfigType:
-    """Validate config."""
-    platform = _get_trigger_platform(config)
```

Comment:
This is the code that we're moving to standardize. Are there other integrations that want to split their trigger code?

Reply:
This will add a 2nd way of doing the same thing. We need to support the old one forever too. I am not sure if this cleanup is going to help make the code easier to understand, as there are now 2 ways of doing it ?

Reply:
The rationale for the PR is that we think there will be a large amount of integrations offering multiple triggers, which means the existing interface is not suitable.
There are only ~10 custom integrations implementing triggers, so if we deprecate the old interface and encourage custom integration authors to update that won't have a very large impact.